src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyCodeFixProvider.cs

```diff
@@ -219,6 +219,19 @@ private async Task<Solution> ProcessResultAsync(CodeFixContext context, Diagnost
             }
         }
 
+        private SyntaxRemoveOptions CreateSyntaxRemoveOptions(SyntaxNode nodeToRemove)
+        {
+            var syntaxRemoveOptions = SyntaxGenerator.DefaultRemoveOptions;
+            var hasDirective = nodeToRemove.GetLeadingTrivia().Any(t => t.IsDirective);
+
+            if (hasDirective)
+            {
+                syntaxRemoveOptions |= SyntaxRemoveOptions.KeepLeadingTrivia;
+            }
+
+            return syntaxRemoveOptions;
```

Comment:
this fix feels odd to me.  I thought this was what KeepUnbalancedDirectives was supposed to do.  It would keep the directive if removing things would lead to an imbalance.

Reply:
If I had used `SyntaxRemoveOptions.KeepUnbalancedDirectives`, it would have added extra EndOfLine character before the directive. I got inspired by: https://github.com/dotnet/roslyn/blob/7854e488100cca6b426bda359bac5e5535d01289/src/Features/Core/Portable/RemoveUnusedVariable/AbstractRemoveUnusedVariableCodeFixProvider.cs#L119-L122

Reply:
> If I had used SyntaxRemoveOptions.KeepUnbalancedDirectives, it would have added extra EndOfLine character before the directive. I got inspired by: 

Would it have added an extra EOL?  Or would it just have preserved what was tthere?  I think having hte code be:

```c#
#region

#endregion
```

is fine...

Reply:
Actually, it is more of this case. 
Before CodeFix:
    
    class Class   
    {
        #region Test
        int i;
        #endregion

After CodeFix:
    
    class Class
    {

        #region Test
        #endregion

It's only case, when it's on the first line in the class space.
Before CodeFix:
 
    class Class   
    {
            
        #region Test
        int i;
        #endregion

After CodeFix:

    class Class
    {
     
        #region Test
        #endregion

It works as expected. But I'm not sure what is proper behavior, so I solved this issue the same way as `RemoveUnusedVariableCodeFix` does.